### PR TITLE
feat(app-platform): Integration "Learn More" modal

### PR DESCRIPTION
**Pull request by mnoble**
Adds a modal for when the User clicks "Learn More" on the Integrations page, for a Sentry App.

image

NOTE: The "By undefined" will say "By \<author>" once https://github.com/getsentry/sentry/pull/12622 is merged and out.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  render() {
+    const {sentryApp, closeModal, onInstall, isInstalled, organization} = this.props;
+
+    return (
+      <React.Fragment>
+        <Flex align="center" mb={2}>
+          <PluginIcon pluginId={sentryApp.slug} size={50} />
+
+          <Flex pl={1} align="flex-start" direction="column" justify="center">
+            <Name>{t(sentryApp.name)}</Name>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx]: I don't think t() can do much here as the POT file extractor will not be able to see these message strings.

File: src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        <Metadata>
+          <Author flex={1}>{t('By %s', sentryApp.author)}</Author>
+        </Metadata>
+
+        <div className="modal-footer">
+          <Button size="small" onClick={closeModal}>
+            {t('Cancel')}
+          </Button>
+
+          <Access organization={organization} access={['org:Integrations']}>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx]:
```suggestion
          <Access organization={organization} access={['org:integrations']}>
```

File: src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+              </Button>
+            )}
+          </Access>
+        </div>
+      </React.Fragment>
+    );
+  }
+}
+
+const Name = styled(p => <Box {...p} />)`
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx]:
```suggestion
const Name = styled(Box)`
```

Should also work.

File: src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    isInstalled: PropTypes.bool.isRequired,
+    closeModal: PropTypes.func.isRequired,
+  };
+
+  render() {
+    const {sentryApp, closeModal, onInstall, isInstalled, organization} = this.props;
+
+    return (
+      <React.Fragment>
+        <Flex align="center" mb={2}>
```
Comment (MeredithAnya) [inline src/sentry/static/sentry/app/components/modals/sentryAppDetailsModal.jsx]: I'm assuming this is copy-pasta from the other modals, but it seems weird that we have some inline styling and then some using the styled components. I feel like they should all be styled components

File: tests/js/spec/views/settings/organizationDeveloperSettings/index.spec.jsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ describe('Organization Developer Settings', function() {
     });
 
     const wrapper = mount(
-      <OrganizationDeveloperSettings params={{orgId: org.slug}} />,
+      <OrganizationDeveloperSettings params={{orgId: org.slug}} organization={org} />,
```
Comment (MeredithAnya) [inline tests/js/spec/views/settings/organizationDeveloperSettings/index.spec.jsx]: do we need the `orgId` as the prop now?

File: tests/js/spec/views/settings/organizationDeveloperSettings/index.spec.jsx [hunk 6]
```diff
@@ -38,7 +38,7 @@ describe('Organization Developer Settings', function() {
...
```
Comment (MeredithAnya) [inline tests/js/spec/views/settings/organizationDeveloperSettings/index.spec.jsx]: same here about the `orgId`

File: tests/js/spec/views/settings/organizationDeveloperSettings/index.spec.jsx [hunk 7]
```diff
@@ -92,7 +92,7 @@ describe('Organization Developer Settings', function() {
       body: [sentryApp],
     });
     const wrapper = mount(
-      <OrganizationDeveloperSettings params={{orgId: newOrg.slug}} />,
+      <OrganizationDeveloperSettings params={{orgId: newOrg.slug}} organization={org} />,
```
Comment (MeredithAnya) [inline tests/js/spec/views/settings/organizationDeveloperSettings/index.spec.jsx]: should `organization={org}` be using the `newOrg`?

**Comment by MeredithAnya** [comment]
Can we add a test for the actual modal itself? Mainly to check if the button is disabled if the integration is already installed.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**